### ISSUE TITLE
TODOs in JavaDoc Comments

### ISSUE THREAD

**Issue by nickPietrass** [Gruntfuggly/todo-tree#111]
Currently TODO's in 
`/**
* TODO
*
**/
`
blocks is not listed. This is fixed by modifying the todo-tree.regex setting to "((//|#|\\*|<!--|;|/\\*|^)\\s*($TAGS)|^\\s*- \\[ \\])". An option to add this by default would be useful. Otherwise a wiki entrance might help visibility especially for new java programmers

**Comment by Gruntfuggly** [comment]
Thanks for the suggestion. It would be nice to support more languages by default, but there are so many different ways that TODOs can be defined that it would be impossible to cater for them all.

The reason the regex configurable is to allow users to customise it for their own use (as you have done).

A wiki page is probably a good idea though - I could make it publicly edittable too, so that people can add their own examples.

**Comment by nickPietrass** [comment]
Oh I must have edited it out, but the style I mentioned is JavaDoc, just in case you want to add it. the GitHub Wiki option should be a good place for alternative regexes since it should rank fairly high on search results by default. Ty for your work

**Comment by LasyIsLazy** [comment]
> Are you sure that "((//|#||<!--|;|/|^)\s*($TAGS)|^\s*- [ ])" works ?

Not work for me, and I didn't see something like `/**` in this regex.

**Comment by LasyIsLazy** [comment]
This is my regex: `(//|#|<!--|/\\*|^\\s*\\*)\\s*($TAGS)`.
It will match
- `//`
- `#`
- `<!--`
- `/*`, 
- `* `(for JSDoc)

**Comment by Gruntfuggly** [comment]
`^\s*` matches start of line, any whitespace and then a star, so it should work in the example in the first post.

**Comment by LasyIsLazy** [comment]
> `^\s*` matches start of line, any whitespace and then a star, so it should work in the example in the first post.

In the first post, it's `^\s*- \[ \])`

**Comment by kanlukasz** [comment]
> 
> 
> This is my regex: `(//|#|<!--|/\\*|^\\s*\\*)\\s*($TAGS)`.
> It will match
> 
>     * `//`
> 
>     * `#`
> 
>     * `<!--`
> 
>     * `/*`,
> 
>     * `* `(for JSDoc)

Thank you so much!!!!
That working brilliant with a lot of languages and types of comments, example: JS, BASH, PHP, HTML, CSS, MD, JSON, XML.  😲 .
Amazing, you save my life. Big  🍺 for you

**Comment by artginzburg** [comment]
> This is my regex: `(//|#|<!--|/\\*|^\\s*\\*)\\s*($TAGS)`. It will match
> 
> * `//`
> * `#`
> * `<!--`
> * `/*`,
> * `* `(for JSDoc)

I think it should also match `/**` — the start of a JSDoc comment, which is sometimes one-line (which is valid by JSDoc specification), like this: `/** TODO rewrite if possible */`.

If we go that way, it's also worth mentioning that JSDoc has its own todo specifier — `@todo`, which would also be good to match, but it needs to only be the `@todo`, without "fixme"/"bug"/etc.

I'll try adding that to the regex and will post here if I succeed.

**Comment by UnknownMasterCoder** [comment]
> This is my regex: (//|#|<!--|/\\*|^\\s*\\*)\\s*($TAGS).

Bro @LasyIsLazy thanks a lot! You saved my life. Such an outstanding contribution for we the mortals who don't are pros with regex

**Comment by swirle13** [comment]
Wanted to throw my hat in the ring. Apologies for commenting on a 4+ year old issue, but it was one of the first results on google for "vscode jsdoc doesn't highlight todo".

### TL;DR: `(//|#|<!--|;|/\*{1,2}|^\s*\*|^|^\s*(-|\d+.))\s*($TAGS)`

---

The default regex was the following: `(//|#|<!--|;|/\*|^|^[ \t]*(-|\d+.))\s*($TAGS)` 

**Note** that this doesn't escape any slash characters. This meant that all the suggestions with double slashes (e.g. `\\s`) did not work for me. I'm on VSCode v1.83.0 as of writing this.

I made a small improvement to @artginzburg's improvement on @LasyIsLazy's solution and integrated the changes from the current updated regex.

```diff
- (//|#|<!--|/(\\*){1,2}|^\\s*\\*)\\s*($TAGS)|(/(\\*){1,2}|^\\s*\\*)\\s*@todo
+ (//|#|<!--|;|/\*{1,2}|^\s*\*|^|^\s*(-|\d+.))\s*($TAGS)
```

1. Removed unnecessary add-on `|(/(\\*){1,2}|^\\s*\\*)\\s*@todo`.  To capture additional tags, these should be added to `"todo-tree.general.tags": [ ... ]` within `settings.json` or via Settings pane with Setting ID `todo-tree.general.tags`
2. Removed unnecessary capture group `/(\\*){1,2}` -> `/\*{1,2}`. It didn't add any additional capturing or grouping for todo-tree task grouping.
3. Changed `^[ \t]*` -> `^\s*` as `\s` already captures spaces and tabs and should capture more whitespace possibilities. I don't see where it would cause issues with newlines as VSCode will understand that to be on a newline, which the caret indicates the regex requires a pattern match at the beginning of the line.

This screenshot shows my small jsdoc tests with this new regex now work (left) as opposed to the default regex (right)
![image](https://github.com/Gruntfuggly/todo-tree/assets/9778474/1d6ca3e8-8e53-40a9-bc1a-e27b84e97794) ![image](https://github.com/Gruntfuggly/todo-tree/assets/9778474/10ad9b4f-0914-4eb6-b60a-053a8b4ff102)
